gitlab-pipeline-checker/wait_for_pipeline.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+
+PROJECT_NAME=$1
+PROJECT_COMMIT_SHA=$2
+
+GITLAB_API_URL=${GITLAB_API_URL:-"https://gitlab.com/api/v4"}
+
+if [ -z "$GITLAB_API_TOKEN" ]; then
+    echo -e "\nError: GITLAB_API_TOKEN variable not set\n"
+    exit 1
+fi
+
+PROJECT_ENCODED=$(echo $PROJECT_NAME | sed -Ee 's/\//%2F/g')
+REQUEST_URL=$GITLAB_API_URL/projects/$PROJECT_ENCODED/repository/commits/$PROJECT_COMMIT_SHA
+
+SLEEP_SECONDS=1
+
+while true; do
+    JSON=$(curl -s --header "PRIVATE-TOKEN: $GITLAB_API_TOKEN" $REQUEST_URL)
+    STATUS=$(echo $JSON | jq -r .last_pipeline.status)
+    if [ "$STATUS" != "running" ] && [ "$STATUS" != "pending" ]; then
+        break
+    fi
+    sleep $SLEEP_SECONDS
+    SLEEP_SECONDS=$((SLEEP_SECONDS + 1))
+done
+
+echo -e "\nPipeline status: $STATUS\n"
```

Comment:
To be more useful to users, can we add the URL to the pipeline here?

Reply:
Good idea, I'm printing the JSON response (only the pipeline) instead now, e.g.:

```json
{
  "id": 111,
  "sha": "aaa",
  "ref": "branch",
  "status": "success",
  "web_url": "https://gitlab.com/project/name/asdf/pipelines/111"
}
```

Is that ok? I don't think it's worth trying to format it in a nice way, this is probably better and can be parsed.

Reply:
Yep, that's okay! Make sure it's piped through jq though to pretty-print it.

Reply:
Yep, done&pushed.